internal/cluster/validator.go

```diff
@@ -305,49 +305,52 @@ func (v *clusterValidator) areIngressVipsValid(c *clusterPreprocessContext) (Val
 	return v.areVipsValid(c, &IngressVipsWrapper{c: c})
 }
 
-// conditions to have a valid number of masters
-// 1. have exactly three masters
-// 2. have less then 3 masters but enough to auto-assign hosts that can become masters
-// 3. have at least 2 workers or auto-assign hosts that can become workers, if workers configured
-// 4. having more then 3 known masters is illegal
-func (v *clusterValidator) sufficientMastersCount(c *clusterPreprocessContext) (ValidationStatus, string) {
+// SufficientMastersCount validates there there is sufficient amount of hosts to satisfy requirments of both masters and workers.
+// The requirments are -
+//   - For none-high-availablity cluster (SNO), exactly 1 master and 0 workers are required.
+//   - For high-availablity cluster, 3-5 masters and any number of workers are required.
+func (v *clusterValidator) SufficientMastersCount(c *clusterPreprocessContext) (ValidationStatus, string) {
 	status := ValidationSuccess
-	var message string
+	var (
+		message                         string
+		minMastersNeededForInstallation int
+		maxMastersNeededForInstallation int
+		mastersRange                    bool
+	)
+
+	isGreaterThan, err := common.BaseVersionGreaterOrEqual(common.MinimumVersionForStretchedControlPlanesCluster, c.cluster.OpenshiftVersion)
```

Comment:
nit: `isGreaterThan` is a bit ambiguous, could we name it something related to the feature about stretch clusters?